internal/controller/pkg/revision/deployment.go

```diff
@@ -78,20 +79,24 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		},
 		ImagePullSecrets: pullSecrets,
 	}
-	secSer := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            *revision.GetTLSServerSecretName(),
-			Namespace:       namespace,
-			OwnerReferences: []metav1.OwnerReference{meta.AsController(meta.TypedReferenceTo(revision, v1.ProviderRevisionGroupVersionKind))},
-		},
+	var secSer, secCli *corev1.Secret
+	if revision.GetTLSServerSecretName() != nil {
+		secSer = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            *revision.GetTLSServerSecretName(),
+				Namespace:       namespace,
+				OwnerReferences: []metav1.OwnerReference{meta.AsController(meta.TypedReferenceTo(revision, v1.ProviderRevisionGroupVersionKind))},
+			},
+		}
 	}
-
-	secCli := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            *revision.GetTLSClientSecretName(),
-			Namespace:       namespace,
-			OwnerReferences: []metav1.OwnerReference{meta.AsController(meta.TypedReferenceTo(revision, v1.ProviderRevisionGroupVersionKind))},
-		},
+	if revision.GetTLSClientSecretName() != nil {
+		secCli = &corev1.Secret{
```

Comment:
I am seeing these being used where `buildProviderDeployment` is called. So we run into different issue there.